### ISSUE TITLE
Strange keybinding, it should start from C rather than E.

### ISSUE THREAD

**Issue by THMonster** [ritiek/piano-rs#23]

**Comment by ritiek** [comment]
> it should start from C rather than E.

I am not sure what you mean by that?

**Comment by THMonster** [comment]
@ritiek  In this project, the computer keyboard key 'z'(or 'q') is mapping to tone E, but in music theory, C is the first tone in every 12 tones circle(called [Equal Temperament](https://en.wikipedia.org/wiki/Equal_temperament)). And piano is a instrument that using Equal Temperament. So the computer keyboard key 'z' and 'q' should mapping to tone C rather then tone E.

**Comment by ritiek** [comment]
I don't know much about music theory (or even piano), hehe. I just copied how http://www.multiplayerpiano.com/ maps keys without going into anything deeper.

If you really want piano-rs to have such mapping, I'd love a PR! We should probably add these new mapping as an optional argument though.

**Comment by THMonster** [comment]
@ritiek In fact, you could simplely refer to VMPK's(a famous cross-platform Virtual MIDI Piano Keyboard) key mapping. And unfortunately, I have no idea about rust, so I can't send a Pull Request.

**Comment by rrooij** [comment]
What about supporting multiple key mappings with a command line argument? That would be pretty neat!

**Comment by ritiek** [comment]
@rrooij Yep, that's what I've been thinking too. I haven't messed anything with this project lately, I'll see what I can do in the coming days though. :smile:

**Comment by ritiek** [comment]
Or if you want to give it a shot, you're most welcome!

**Comment by rrooij** [comment]
I wouldn't mind doing it. I don't have a lot of time, so I can't do it if you're in a hurry, but I was searching on something to practice Rust a little bit with and would like to contribute.

**Comment by ritiek** [comment]
Sure, take your time! Just so you know, the code quality isn't really great or anything. I myself wrote this while learning Rust. :D

**Comment by rrooij** [comment]
Cool! I will keep it in mind.

One cool approach I was thinking of is putting the keyboard layouts in a layouts.yml file so the user can define their own layouts if necessary too.

**Comment by poly000** [comment]
> What about supporting multiple key mappings with a command line argument? That would be pretty neat!

may a configuration file be more appropriate?
